### PR TITLE
CI: Test minimum versions

### PR DESCRIPTION
**Pull request by effigies**
Realized while replacing deprecated calls in #204 that there's the potential to break things by moving to the latest APIs. Adding a CI job to test on explicit minimum requirements.

To make this work, there need to be minimum requirements, so I followed [SPEC 0](https://scientific-python.org/specs/spec-0000/) (follow-up to NEP29), which basically says to drop any minor release series that started more than 2 years ago. At Nibabel we follow NEP29+1y. Added a script to make keeping these up-to-date easy.

If this is overkill, please feel free to close.

### CODE REVIEW

File: tools/update_requirements.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env python3
+import sys
+from pathlib import Path
+
+import tomli
```
Comment (arokem) [inline tools/update_requirements.py:5]: Maybe add tomli to requirements-dev.txt?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -34,12 +38,12 @@ jobs:
...
+        python -m pip install -r ${{ matrix.requires }}
         python -m pip install -r requirements-dev.txt
         python -m pip install .
     - name: Lint
       run: |
         pipx run flake8 --ignore N802,N806,W504 --select W503 nitime/ tools/
     - name: Test
       run: |
-        cd && mkdir for_test && cd for_test && pytest --pyargs nitime --cov-report term-missing --cov=AFQ
+        mkdir ~/for_test && cd ~/for_test && pytest --pyargs nitime --cov-report term-missing --cov=nitime
```
Comment (arokem) [inline .github/workflows/test.yml:49]: Ha! Sorry about that 😄

**Review by arokem — COMMENTED** [review]
Seems helpful and the opposite of overkill (supporting older versions is the overkill). Maybe just a small addition to the development requirements and we can merge this.

File: tools/update_requirements.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python3
+import sys
+from pathlib import Path
+
+try:
+    import tomllib
+except ImportError:
+    import tomli as tomllib
```
Comment (effigies) [inline tools/update_requirements.py:8]: Preferring the builtin for recent Python.

Comment (arokem) [reply tools/update_requirements.py:8]: Nice!
